contrib/uriparser/utils/utils_uriparser.cc

```diff
@@ -0,0 +1,286 @@
+    inet_ntop(AF_INET, ip4.mutable_data()->data, ipstr, sizeof(ipstr));
+  } else if (uri_.mutable_data()->hostData.ip6) {
+    sapi::v::Struct<UriIp6> ip6;
+    ip6.SetRemote(uri_.mutable_data()->hostData.ip6);
+    SAPI_RETURN_IF_ERROR(sandbox_->TransferFromSandboxee(&ip6));
+    inet_ntop(AF_INET6, ip6.mutable_data()->data, ipstr, sizeof(ipstr));
+  }
+
+  return std::string(ipstr);
+}
+
+absl::StatusOr<std::string> Uriparser::GetPortText() {
+  return FetchUriText(&uri_.mutable_data()->portText);
+}
+
+absl::StatusOr<std::string> Uriparser::GetQuery() {
+  return FetchUriText(&uri_.mutable_data()->query);
+}
+
+absl::StatusOr<std::string> Uriparser::GetFragment() {
+  return FetchUriText(&uri_.mutable_data()->fragment);
+}
+
+absl::StatusOr<std::vector<std::string>> Uriparser::GetPath() {
+  std::vector<std::string> ret;
+
+  UriPathSegmentA* pathHead = uri_.mutable_data()->pathHead;
+  if (pathHead == nullptr) {
+    return ret;
+  }
+
+  sapi::v::Struct<UriPathSegmentA> path_segment;
+  path_segment.SetRemote(pathHead);
+
+  while (path_segment.GetRemote() != nullptr) {
+    SAPI_RETURN_IF_ERROR(sandbox_->TransferFromSandboxee(&path_segment));
+
+    SAPI_ASSIGN_OR_RETURN(std::string seg,
+                          FetchUriText(&path_segment.mutable_data()->text));
+    if (!seg.empty()) {
+      ret.push_back(seg);
+    }
+
+    path_segment.SetRemote(path_segment.mutable_data()->next);
+  }
+
+  return ret;
+}
+
+absl::Status Uriparser::NormalizeSyntax() {
+  SAPI_ASSIGN_OR_RETURN(int dirty_parts,
+                        api_.uriNormalizeSyntaxMaskRequiredA(uri_.PtrNone()));
+
+  return NormalizeSyntax(dirty_parts);
+}
+
+absl::Status Uriparser::NormalizeSyntax(int norm_mask) {
+  SAPI_ASSIGN_OR_RETURN(int ret,
+                        api_.uriNormalizeSyntaxExA(uri_.PtrAfter(), norm_mask));
+
+  if (ret != 0) {
+    return absl::UnavailableError("Unable to normalize");
+  }
+
+  return absl::OkStatus();
+}
+
+absl::StatusOr<std::string> Uriparser::GetURI() { return GetURI(&uri_); }
+
+absl::StatusOr<std::string> Uriparser::GetURI(sapi::v::Struct<UriUriA>* uri) {
+  sapi::v::Int size;
+  int ret;
+
+  SAPI_ASSIGN_OR_RETURN(
+      ret, api_.uriToStringCharsRequiredA(uri->PtrNone(), size.PtrAfter()));
+  if (ret != 0) {
+    return absl::UnavailableError("Unable to get size");
+  }
+
+  sapi::v::Array<char> buf(size.GetValue() + 1);
+  sapi::v::NullPtr null_ptr;
+
+  SAPI_ASSIGN_OR_RETURN(ret, api_.uriToStringA(buf.PtrAfter(), uri->PtrNone(),
+                                               buf.GetSize(), &null_ptr));
+  if (ret != 0) {
+    return absl::UnavailableError("Unable to Recomposing URI");
+  }
+
+  return std::string(buf.GetData());
+}
+
+absl::StatusOr<std::string> Uriparser::GetURIEscaped(bool space_to_plus,
+                                                     bool normalize_breaks) {
+  SAPI_ASSIGN_OR_RETURN(std::string uri, GetURI());
+
+  // Be sure to allocate <b>6 times</b> the space of the input buffer for
```

Comment:
We do not use HTML in comments. They are not actually rendered properly (at Google anyways). I'd use Markdown-ish **bold** and _italic_ for emphasis or else skip the markup altogether.

Reply:
This is copy-paste from the documentation, to explain why we are doing that.